### DISCUSSION TITLE
Appender: support column default values

### DISCUSSION THREAD

**Discussion by stevenpelley** [duckdb/duckdb#9158] [Ideas]
there is no mention of column default values in the appender documentation https://duckdb.org/docs/data/appender and no way to explicitly indicate to use the default value within the c api https://duckdb.org/docs/api/c/appender.  I saw no other discussions regarding appender and "default."
INSERT uses the default value when the column is omitted, but this cannot be done with appender.  This makes it impossible to use sequences to generate unique ids while using the appender.
Please let me know if I've overlooked something or if this functionality exists but isn't properly documented (in which case this request is to document it)

I request adding a c function duckdb_append_default() to indicate that a row's column should use its default value.  If any row uses a default value for a column then all rows must use a default value or else an error will be returned either on the first call call to duckdb_append_default or the first call to append a non-default value for a column.

Other language APIs, especially with generic-typed AppendRow functions, may instantiate a token object representing the default value, and callers may pass this object to AppendRow.

This can also be done by indicating to the appender which columns to insert into (omitting columns that should use a default value) or by naming the columns (by name or ordinal) to use default values prior to any call to append.  I think this is a larger change to the appender API though.

**Comment by RobinQu** [comment]
+1, really needed.

**Comment by Tishj** [comment]
https://github.com/duckdb/duckdb/pull/11905
